### PR TITLE
Cleanup BinCompat TODO in System.Private.Xml

### PR DESCRIPTION
**Pull request by sepidehMS**
Resolves https://github.com/dotnet/corefx/issues/9913 (cleaning up leftover TODOs from initial port of Schema and XSLT)

@danmosemsft @weshaggard @stephentoub @krwq

### CODE REVIEW

File: src/System.Private.Xml/src/System/Xml/Xsl/IlGen/XmlILModule.cs [hunk 1]
```diff
@@ -138,10 +135,7 @@ public XmlILModule(bool useLRE, bool emitSymbols)
                 }
 
                 // Create ModuleBuilder
-                if (_persistAsm)
-                    modBldr = asmBldr.DefineDynamicModule("System.Xml.Xsl.CompiledQuery");
-                else
-                    modBldr = asmBldr.DefineDynamicModule("System.Xml.Xsl.CompiledQuery"); //, this.modFile + ".dll", emitSymbols); //BinCompat TODO
```
Comment (danmosemsft) [inline src/System.Private.Xml/src/System/Xml/Xsl/IlGen/XmlILModule.cs]: >, this.modFile + ".dll", emitSymbols) [](start = 93, length = 37)

odd I would expect these parameters to be used if _persistAsm is TRUE

Comment (sepidehMS) [inline src/System.Private.Xml/src/System/Xml/Xsl/IlGen/XmlILModule.cs]: You are absolutely right. I noticed this was switched by myself mistakenly when I was porting this. You can see the Desktop version is the other way: http://index/?rightProject=System.Data.SqlXml&file=System%5cXml%5cXsl%5cIlGen%5cXmlILModule.cs&line=150

**Comment by danmosemsft** [comment]
given therea re no test changes in this commit I'm guessing there is no test coverage for this stuff? fine for now.

**Comment by danmosemsft** [comment]
:shipit:

**Pull request merged**